File: project/changzune-onboarding/src/main/java/com/innercircle/onboarding/changzune_onboarding/survey/dto/SurveyResponse.java
package com.innercircle.onboarding.changzune_onboarding.survey.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.List;

@Getter
@AllArgsConstructor
public class SurveyResponse {
    private Long id;
    private String title;
    private String description;
    private List<QuestionDto> questions;

    @Getter
    @AllArgsConstructor
    public static class QuestionDto {
        private Long id;

        private String name;
        private String description;
        private String type;
        private boolean required;
        private List<String> options;
    }
}

File: project/jinho-onboarding/src/main/kotlin/fc/innercircle/jinhoonboarding/survey/dto/response/SurveyResponse.kt
package fc.innercircle.jinhoonboarding.survey.dto.response

import fc.innercircle.jinhoonboarding.survey.domain.Survey

data class SurveyResponse(
    val title: String,
    val description: String,
    val questions: List<QuestionResponse>
){
    data class QuestionResponse (
        val title: String,
        val description: String,
        val questionType: String,
        val required: Boolean = false,
        val options: List<String> = emptyList()
    )

    companion object {
        fun fromDomain(response: Survey): SurveyResponse {
            val questions = response.questions.map {
                QuestionResponse(
                    title = it.title,
                    description = it.description,
                    questionType = it.questionType.name,
                    required = it.required,
                    options = it.options
                )
            }.toList()

            val surveyResponse = SurveyResponse(
                title = response.title,
                description = response.description,
                questions = questions
            )

            return surveyResponse
        }
    }

}